doc/rst/technotes/initequals.rst

```diff
@@ -108,21 +108,38 @@ accepts one argument of the same type:
 
 In order to override this compiler-generated implementation, the user must
 implement an ``init=`` method that can accept an argument of the same type.
-Other user-defined ``init=`` methods will not prevent the compiler from
-generating a default implementation for ``init=``. For example:
+
+An ``init=`` method may also specify a type other than the one being
+initialized. Such user-defined initializers will disable generation of the
+default implementation of a type's ``init=`` method, and users must provide an
+equivalent implementation.
+
+When initializing from a different type, users must also provide a cast
+implementation between their record and the other type. This requirement exists
+for consistency and completeness, as both ``init=`` and a cast create one type
+from another. For example, consider a record that can be initialized from an
+``int``.
```

Comment:
With the "For example" sentence in this paragraph, I expected the example to suggest why the cast operator is needed for consistency.

Reply:
@mppf : Do you have any recommendations for language here to explain the cast requirement? Alternatively, I could just turn "For example, consider" into "Consider".

Reply:
Hmm.. see https://chapel-lang.org/releaseNotes/1.24/01-lang-lib.pdf slide 15. But I'm not sure that has a 1-sentence summary. I think the way I would summarize it is:
 1. It allows the compiler more freedom to use the appropriate operation (e.g. with split-init, it chooses an `init=` instead of an `=`)
 2. It makes the convert-ability between two types be one of 4 levels instead of being different for each operation. This helps to keep types consistent and enables Chapel programmers to make simplifying assumptions (e.g. I saw an `=` between two types in some code; therefore I must also be able to explicitly cast with `:` between those types).

I am not so sure that this level of detail needs to go into this paragraph, though.

Reply:
I just pushed a change — does it look good to you both?